ndindex/_tuple.pyx

```diff
@@ -72,7 +79,7 @@ cdef void _lazy_import():
 
 cdef int _is_boolean_scalar(object idx):
     cdef object BooleanArray
-    _lazy_import()
+    call_once(_lazy_import_lock, _lazy_import)
```

Comment:
To avoid a possible deadlock with the interpreter, I think you need to do this:

```
with nogil:
    call_once(_lazy_import_lock, _lazy_import)
```

Unless you know somehow that the GIL has already been released in every caller?

Reply:
Ah, no you're right, can't assume that.. updating.